plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -211,6 +211,12 @@ public void registerTable(ConnectorSession session, SchemaTableName tableName, S
         throw new TrinoException(NOT_SUPPORTED, "registerTable is not supported for Iceberg REST catalog");
     }
 
+    @Override
+    public void unregisterTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "unregisterTable is not supported for Iceberg REST catalogs");
```

Comment:
Same here, could do something like:
```
restSessionCatalog.dropTable(convert(session), toIdentifier(schemaTableName));
```

I think there's a related bug in this `dropTable` implementation below. There's a separate method called `RESTSessionCatalog#purgeTable` for dropping a table when the files should be deleted I think we should call instead.

Reply:
This is not a bug in the current version because the method calls `purgeTable` method internally. Sent https://github.com/trinodb/trino/pull/15855